include/sys/vm_object.h

```diff
@@ -14,12 +16,13 @@
  */
 
 typedef struct vm_object {
-  mtx_t mtx;
+  rwlock_t mtx;
   vm_pagelist_t list;   /* (@) List of pages */
   vm_pagetree_t tree;   /* (@) Tree of pages */
   size_t npages;        /* (@) Number of pages */
   vm_pager_t *pager;    /* Pager type and page fault function for object */
   refcnt_t ref_counter; /* (a) How many objects refer to this object? */
+  vm_object_t *shadow_object; /* (@) Pointer to backing object */
```

Comment:
Maybe some prefix like `vmo_` will be a good idea here.

Reply:
Do you mean adding to each field?

Reply:
Yes.